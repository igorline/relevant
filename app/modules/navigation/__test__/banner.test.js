import React from 'react';
import renderer from 'react-test-renderer';
import Banner from 'modules/navigation/banner';
import 'jest-styled-components';
import { MemoryRouter } from 'react-router-dom';

const props = {
  cta: 'SIGN_UP'
};

jest.mock('react-redux', () => {
  const state = {
    navigation: { screenSize: 0 }
  };
  return {
    useDispatch: () => {},
    connect: el => el,
    useSelector: fn => fn(state)
  };
});

test('Snapshot SplashComponent', () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <Banner {...props} overRideDismiss />
      </MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
