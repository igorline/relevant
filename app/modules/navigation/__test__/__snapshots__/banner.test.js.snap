// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot SplashComponent 1`] = `
.c0 {
  color: #0000ff;
}

.c1 {
  color: #0000ff;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

<div
  className="css-view-1dbjc4n"
  style={
    Object {
      "WebkitAlignItems": "center",
      "WebkitBoxAlign": "center",
      "WebkitBoxDirection": "normal",
      "WebkitBoxOrient": "horizontal",
      "WebkitBoxPack": "start",
      "WebkitFlexDirection": "row",
      "WebkitJustifyContent": "flex-start",
      "alignItems": "center",
      "borderBottomColor": "rgba(222,222,222,1.00)",
      "borderBottomStyle": "solid",
      "borderBottomWidth": "1px",
      "borderLeftStyle": "solid",
      "borderRightStyle": "solid",
      "borderTopStyle": "solid",
      "display": "flex",
      "flexDirection": "row",
      "justifyContent": "flex-start",
      "maxHeight": "550px",
      "msFlexAlign": "center",
      "msFlexDirection": "row",
      "msFlexPack": "start",
      "overflowX": "hidden",
      "overflowY": "hidden",
      "position": "relative",
    }
  }
>
  <div
    className="css-view-1dbjc4n"
    data-focusable={true}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "MozTransitionProperty": "opacity",
        "MozUserSelect": "none",
        "WebkitTransitionDuration": "0.15s",
        "WebkitTransitionProperty": "opacity",
        "WebkitUserSelect": "none",
        "cursor": "pointer",
        "msTouchAction": "manipulation",
        "msUserSelect": "none",
        "opacity": 1,
        "touchAction": "manipulation",
        "transitionDuration": "0.15s",
        "transitionProperty": "opacity",
        "userSelect": "none",
      }
    }
    tabIndex="0"
  >
    <div
      className="css-view-1dbjc4n"
      style={
        Object {
          "MozUserSelect": "none",
          "WebkitFlexBasis": "auto",
          "WebkitUserSelect": "none",
          "cursor": "pointer",
          "display": "flex",
          "flexBasis": "auto",
          "height": "1.5rem",
          "msFlexPreferredSize": "auto",
          "msUserSelect": "none",
          "overflowX": "hidden",
          "overflowY": "hidden",
          "position": "absolute",
          "right": "3rem",
          "top": "3rem",
          "userSelect": "none",
          "width": "1.5rem",
          "zIndex": 10,
        }
      }
    >
      <div
        className="css-view-1dbjc4n"
        style={
          Object {
            "backgroundColor": "rgba(0,0,0,0.00)",
            "backgroundImage": "url(\\"x.png\\")",
            "backgroundPosition": "center",
            "backgroundRepeat": "no-repeat",
            "backgroundSize": "contain",
            "bottom": "0px",
            "height": "100%",
            "left": "0px",
            "position": "absolute",
            "right": "0px",
            "top": "0px",
            "width": "100%",
            "zIndex": -1,
          }
        }
      />
      <img
        alt=""
        className="css-accessibilityImage-9pa8cd"
        draggable={false}
        src="x.png"
      />
    </div>
  </div>
  <div
    className="css-view-1dbjc4n"
    style={
      Object {
        "WebkitAlignItems": "flex-start",
        "WebkitBoxAlign": "start",
        "WebkitBoxDirection": "normal",
        "WebkitBoxFlex": 1,
        "WebkitBoxOrient": "vertical",
        "WebkitBoxPack": "center",
        "WebkitFlexBasis": "0px",
        "WebkitFlexDirection": "column",
        "WebkitFlexGrow": 1,
        "WebkitFlexShrink": 1,
        "WebkitJustifyContent": "center",
        "alignItems": "flex-start",
        "display": "flex",
        "flexBasis": "0px",
        "flexDirection": "column",
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
        "marginBottom": "3rem",
        "marginLeft": "6rem",
        "marginRight": "0rem",
        "marginTop": "3rem",
        "msFlexAlign": "start",
        "msFlexDirection": "column",
        "msFlexNegative": 1,
        "msFlexPack": "center",
        "msFlexPositive": 1,
        "msFlexPreferredSize": "0px",
        "position": "relative",
      }
    }
  >
    <div
      className="css-view-1dbjc4n"
      style={
        Object {
          "display": "flex",
          "position": "relative",
        }
      }
    >
      <div
        className="css-text-901oao"
        dir="auto"
        style={
          Object {
            "color": "rgba(0,0,0,1.00)",
            "fontFamily": "HelveticaNeue-Medium",
            "fontSize": "3rem",
            "lineHeight": "4.5rem",
          }
        }
      >
        <span
          className="css-text-901oao css-textHasAncestor-16my406"
          dir="auto"
          style={
            Object {
              "display": "flex",
              "position": "relative",
            }
          }
        >
          Information curated by humans, not algorithms.
        </span>
      </div>
      <div
        className="css-view-1dbjc4n"
        style={
          Object {
            "display": "flex",
            "marginBottom": "3rem",
            "marginTop": "2.5rem",
            "position": "relative",
          }
        }
      >
        <div
          className="css-text-901oao"
          dir="auto"
          style={
            Object {
              "display": "inline",
              "fontFamily": "Georgia",
              "fontSize": "1.25rem",
              "lineHeight": "2rem",
              "position": "relative",
            }
          }
        >
          Join a 
          <a
            className="c0 ulink-mock"
            href="/communities"
            hu={0}
            inline={0}
            onClick={[Function]}
            styles=""
          >
            community
          </a>
          , curate content and earn rewards.
           
          <a
            className="c1 ulink-mock"
            display="inline"
            href="https://blog.relevant.community/relevant-curated-by-communities-not-clicks-ba8d346c47da"
            onClick={[Function]}
            target="_blank"
          >
            Learn More
          </a>
        </div>
      </div>
    </div>
    <div
      className="css-view-1dbjc4n"
      style={
        Object {
          "WebkitBoxDirection": "normal",
          "WebkitBoxOrient": "horizontal",
          "WebkitBoxPack": "start",
          "WebkitFlexDirection": "row",
          "WebkitJustifyContent": "flex-start",
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "msFlexDirection": "row",
          "msFlexPack": "start",
          "position": "relative",
        }
      }
    >
      <div
        className="css-view-1dbjc4n"
        data-focusable={true}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "MozTransitionProperty": "opacity",
            "MozUserSelect": "none",
            "WebkitTransitionDuration": "0.15s",
            "WebkitTransitionProperty": "opacity",
            "WebkitUserSelect": "none",
            "cursor": "pointer",
            "msTouchAction": "manipulation",
            "msUserSelect": "none",
            "opacity": 1,
            "touchAction": "manipulation",
            "transitionDuration": "0.15s",
            "transitionProperty": "opacity",
            "userSelect": "none",
          }
        }
        tabIndex="0"
      >
        <div
          className="css-view-1dbjc4n"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          style={
            Object {
              "MozUserSelect": "none",
              "WebkitAlignItems": "center",
              "WebkitBoxAlign": "center",
              "WebkitBoxPack": "center",
              "WebkitJustifyContent": "center",
              "WebkitUserSelect": "none",
              "alignItems": "center",
              "backgroundColor": "rgba(0,0,255,1.00)",
              "cursor": "pointer",
              "display": "flex",
              "height": "3rem",
              "justifyContent": "center",
              "marginBottom": "0px",
              "marginLeft": "0px",
              "marginRight": "2rem",
              "marginTop": "0px",
              "minWidth": "9rem",
              "msFlexAlign": "center",
              "msFlexPack": "center",
              "msUserSelect": "none",
              "paddingBottom": "0px",
              "paddingLeft": "1rem",
              "paddingRight": "1rem",
              "paddingTop": "0px",
              "position": "relative",
              "userSelect": "none",
            }
          }
        >
          <div
            className="css-text-901oao"
            dir="auto"
            style={
              Object {
                "color": "rgba(255,255,255,1.00)",
                "fontFamily": "HelveticaNeue-Medium",
                "fontSize": "0.75rem",
              }
            }
          >
            Login
          </div>
        </div>
      </div>
      <div
        className="css-view-1dbjc4n"
        data-focusable={true}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "MozTransitionProperty": "opacity",
            "MozUserSelect": "none",
            "WebkitTransitionDuration": "0.15s",
            "WebkitTransitionProperty": "opacity",
            "WebkitUserSelect": "none",
            "cursor": "pointer",
            "msTouchAction": "manipulation",
            "msUserSelect": "none",
            "opacity": 1,
            "touchAction": "manipulation",
            "transitionDuration": "0.15s",
            "transitionProperty": "opacity",
            "userSelect": "none",
          }
        }
        tabIndex="0"
      >
        <div
          className="css-view-1dbjc4n"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          style={
            Object {
              "MozUserSelect": "none",
              "WebkitAlignItems": "center",
              "WebkitBoxAlign": "center",
              "WebkitBoxPack": "center",
              "WebkitJustifyContent": "center",
              "WebkitUserSelect": "none",
              "alignItems": "center",
              "backgroundColor": "rgba(0,0,255,1.00)",
              "cursor": "pointer",
              "display": "flex",
              "height": "3rem",
              "justifyContent": "center",
              "marginBottom": "0px",
              "marginLeft": "0px",
              "marginRight": "0px",
              "marginTop": "0px",
              "minWidth": "9rem",
              "msFlexAlign": "center",
              "msFlexPack": "center",
              "msUserSelect": "none",
              "paddingBottom": "0px",
              "paddingLeft": "1rem",
              "paddingRight": "1rem",
              "paddingTop": "0px",
              "position": "relative",
              "userSelect": "none",
            }
          }
        >
          <div
            className="css-text-901oao"
            dir="auto"
            style={
              Object {
                "color": "rgba(255,255,255,1.00)",
                "fontFamily": "HelveticaNeue-Medium",
                "fontSize": "0.75rem",
              }
            }
          >
            Sign Up
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="css-view-1dbjc4n"
    style={
      Object {
        "WebkitAlignSelf": "flex-end",
        "WebkitBoxFlex": 0.65,
        "WebkitFlexBasis": "0px",
        "WebkitFlexGrow": 0.65,
        "WebkitFlexShrink": 1,
        "WebkitTransformOrigin": "bottom",
        "alignSelf": "flex-end",
        "display": "flex",
        "flexBasis": "0px",
        "flexGrow": 0.65,
        "flexShrink": 1,
        "maxHeight": "400px",
        "msFlexItemAlign": "end",
        "msFlexNegative": 1,
        "msFlexPositive": 0.65,
        "msFlexPreferredSize": "0px",
        "position": "relative",
        "transformOrigin": "bottom",
        "zIndex": -1,
      }
    }
  >
    <img
      alt="phone"
      onLoad={[Function]}
      src="/img/hand-transparent.png"
      style={
        Object {
          "width": "100%",
        }
      }
    />
  </div>
</div>
`;
